frontend/src/router/index.js

```diff
@@ -190,40 +190,45 @@ router.beforeEach((to, _from, next) => {
 
   function validateAndExecute(nextRouteInError, to) {
     const authStore = useAuthStore()
-    authStore
-      .getJwtToken()
-      .then(() => {
-        if (!authStore.isAuthenticated) {
-          next(nextRouteInError)
-          return
-        }
-        if (!to.meta.role) {
-          next()
-          return
-        }
-        authStore
-          .getUserInfo()
-          .then(() => {
-            if (!authStore.isAuthorizedUser) {
-              next('unauthorized')
-              return
-            }
-            const hasRole = Object.prototype.hasOwnProperty.call(authStore, to.meta.role) && authStore[to.meta.role]
-            if (!hasRole) {
-              next('unauthorized-page')
-              return
-            }
-            next()
-          })
-          .catch((e) => {
-            console.log('Unable to get user info: ' + e)
-            next('error')
-          })
-      })
-      .catch(() => {
-        console.log('Unable to get token')
-        next(nextRouteInError)
-      })
+    console.log('router.validateAndExecute  1')
+    if (to.meta.requiresAuth) {
+      console.log('router.validateAndExecute  2')
+      authStore
+        .getJwtToken()
+        .then(() => {
+          if (!authStore.isAuthenticated) {
+            next('/token-expired')
```

Comment:
Instead of an else block this code might be better with a return statement after the next().

Reply:
Confirming completed.